queries.sql

```diff
@@ -279,5 +279,121 @@ GROUP BY
   owners.full_name
 ORDER BY
   animal_count DESC
+LIMIT
+  1;
+
+-- Who was the last animal seen by William Tatcher?
+SELECT
+  animals.name
+FROM
+  animals
+  JOIN visits ON animals.id = visits.animal_id
+  JOIN vets ON visits.vet_id = vets.id
+WHERE
+  vets.name = 'William Tatcher'
+ORDER BY
+  visits.visit_date DESC
+LIMIT
+  1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT
+  COUNT(DISTINCT visits.animal_id) AS animal_count
+FROM
+  visits
+  JOIN vets ON visits.vet_id = vets.id
+WHERE
+  vets.name = 'Stephanie Mendez';
+
+-- List all vets and their specialties, including vets with no specialties.
+SELECT
+  vets.name,
+  COALESCE(species.name, 'No specialty') AS specialty
+FROM
+  vets
+  LEFT JOIN specializations ON vets.id = specializations.vet_id
+  LEFT JOIN species ON specializations.species_id = species.id;
+
+-- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT
+  animals.name
+FROM
+  animals
+  JOIN visits ON animals.id = visits.animal_id
+  JOIN vets ON visits.vet_id = vets.id
+WHERE
+  vets.name = 'Stephanie Mendez'
+  AND visits.visit_date >= '2020-04-01'
+  AND visits.visit_date <= '2020-08-30';
+
+-- What animal has the most visits to vets?
+SELECT
+  animals.name,
+  COUNT(*) AS visit_count
+FROM
+  animals
+  JOIN visits ON animals.id = visits.animal_id
+GROUP BY
+  animals.name
+ORDER BY
+  visit_count DESC
+LIMIT
+  1;
+
+-- Who was Maisy Smith's first visit?
+SELECT
+  animals.name
+FROM
+  animals
+  JOIN visits ON animals.id = visits.animal_id
+  JOIN vets ON visits.vet_id = vets.id
+WHERE
+  vets.name = 'Maisy Smith'
+ORDER BY
+  visits.visit_date ASC
+LIMIT
+  1;
+
+-- Details for most recent visit animal information, vet information, and date of visit.
+SELECT
+  animals.name AS animal_name,
+  vets.name AS vet_name,
+  visits.visit_date
+FROM
+  visits
+  JOIN animals ON visits.animal_id = animals.id
+  JOIN vets ON visits.vet_id = vets.id
+ORDER BY
+  visits.visit_date DESC
+LIMIT
+  1;
+
+-- How many visits were with a vet that did not specialize in that animal's species?
+SELECT
+  COUNT(*) AS mismatch_count
+FROM
+  visits
+  JOIN animals ON visits.animal_id = animals.id
+  JOIN vets ON visits.vet_id = vets.id
+  LEFT JOIN specializations ON vets.id = specializations.vet_id
+  AND animals.species_id = specializations.species_id
+WHERE
+  specializations.vet_id IS NULL;
+
+-- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+SELECT
+  species.name AS specialty
+FROM
+  visits
+  JOIN animals ON visits.animal_id = animals.id
+  JOIN vets ON visits.vet_id = vets.id
+  JOIN specializations ON vets.id = specializations.vet_id
+  JOIN species ON specializations.species_id = species.id
+WHERE
+  vets.name = 'Maisy Smith'
+GROUP BY
+  species.name
+ORDER BY
+  COUNT(*) DESC
```

Comment:
- [ ] Kindly consider correcting this query as this results in a wrong result. Maisy Smith should consider getting `Digimon` as she got 6 visits of that species.

Reply:
Hi @Zuhaib042, Thanks for bringing this to my attentions😊.  I have updated the query.